llvm/lib/Transforms/InstCombine/InstCombineMulDivRem.cpp

```diff
@@ -1544,6 +1544,10 @@ Instruction *InstCombinerImpl::visitSDiv(BinaryOperator &I) {
     }
   }
 
+  // -(X * Y) / (X * Y) --> -1
+  if (isKnownNegation(Op0, Op1, true)) {
+    return replaceInstUsesWith(I, ConstantInt::getAllOnesValue(Ty));
+  };
```

Comment:
```suggestion
```

It has been handled by `simplifySDivInst`.
https://github.com/llvm/llvm-project/blob/daddf402d9a4e3bacc37098d990e56dc9957ca3e/llvm/lib/Analysis/InstructionSimplify.cpp#L1230-L1232